### PR TITLE
Entities & hooks fix-ups/refactor

### PR DESCRIPTION
**Pull request by Polatrite**

### CODE REVIEW

File: src/client/entities/player.client.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+				graphics: entity.sprite
+			}
+        },
+        delete({ nid, entity, graphics }) {
+			GameRenderer.instance.backgroundLayer.removeChild(graphics)
+        },
+        watch: {
+			// propName: ({ value, entity, graphics }) => {
+			// 	// do thing
+			// },
```
Comment (mdonjones) [inline src/client/entities/player.client.ts:33]: 🔪

File: src/client/networking/client-hooks.ts [hunk 2]
```diff
@@ -45,91 +45,86 @@ export function clientHooks(client, hooks) {
...
-    client.on('update', update => {
+    client.on('update', (update) => {
         if (client.entityUpdateFilter && client.entityUpdateFilter(update)) {
             //console.log('ignore', update)
             return
         }
         const entity = client.entities.get(update.nid)
+        const name = entity.protocol.name
         if (entity) {
+            // console.log(`[${update.nid}] updated ${update.prop} from ${entity[update.prop]} to ${update.value}`)
```
Comment (mdonjones) [inline src/client/networking/client-hooks.ts:92]: I would either 🔪 this if it's unintended, or just leave uncommented as `console.info` or something

File: src/client/networking/client-hooks.ts [hunk 3]
```diff
@@ -45,91 +45,86 @@ export function clientHooks(client, hooks) {
...
             entity[update.prop] = update.value
         } else {
-            console.log('tried to update a sim that did not exist')
+            console.log(`Hooks tried to update an entity that did not exist: ${update.nid}`)
```
Comment (mdonjones) [inline src/client/networking/client-hooks.ts:93]: Same here. If we make it `console.info` we can at least filter it out when we don't want to see it

File: src/client/networking/client-hooks.ts [hunk 4]
```diff
@@ -45,91 +45,86 @@ export function clientHooks(client, hooks) {
...
-            const name = graphics.protocol.name
-            const hooks = client.hooks[name]
-
-            if (hooks.watch && hooks.watch[update.prop]) {
-                hooks.watch[update.prop]({ id: update.id, value: update.value, entity, graphics })
-            }
-
         } else {
-            console.log('tried to update an entity that did not exist')
+            console.log(`Hooks tried to update a graphical entity that did not exist: ${update.nid}`)
```
Comment (mdonjones) [inline src/client/networking/client-hooks.ts]: same same

File: src/server/networking/nengi-server.ts [hunk 5]
```diff
@@ -3,12 +3,19 @@ import nengi from 'nengi'
...
 export class NengiServer {
     public static instance: nengi.Instance
 
+    public static entities: Map<any, any>
+
     static update(delta, tick, now) {
         NengiServer.instance.emitCommands()
+
+        this.entities.forEach(entity => {
+        })
```
Comment (mdonjones) [inline src/server/networking/nengi-server.ts]: 🔪 ?

**Review by mdonjones — CHANGES_REQUESTED** [review]

File: src/client/entities/player.client.ts (hunk 1 above)
Comment (Polatrite) [reply src/client/entities/player.client.ts:33]: This is an untyped hook being run by nengi with a weird set of parameters, so I'd prefer to leave the example in until we use it a few times.

File: src/client/networking/client-hooks.ts (hunk 2 above)
Comment (Polatrite) [reply src/client/networking/client-hooks.ts:92]: Having used nengi for a long time, you wind up using this console.log every 1-3 months when somebody forgets how to nengi properly and can't figure out why XYZ isn't happening.

File: src/client/networking/client-hooks.ts (hunk 4 above)
Comment (Polatrite) [reply src/client/networking/client-hooks.ts]: Converting these to the appropriate logger.error/warn messages.

**Pull request merged**